java/cuvs-java/src/main/java22/com/nvidia/cuvs/spi/OptionalNativeDependencyLoader.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.cuvs.spi;
+
+import java.io.*;
+import java.net.URL;
+import java.util.stream.*;
+
+/**
+ * A class that loads native dependencies if they are available in the jar.
+ */
+public class OptionalNativeDependencyLoader {
+
+  private static final ClassLoader loader = JDKProvider.class.getClassLoader();
+
+  private static boolean loaded = false;
+
+  private static String[] filesToLoad = {
```

Comment:
Nit: this can be final. 
Also, following the convention of the rest of the codebase, I would rename it to `LIBRARIES_TO_LOAD`